### DISCUSSION TITLE
CRA 5.0.0 setup doesn't work

### DISCUSSION THREAD

**Discussion by srcreigh** [tailwindlabs/tailwindcss#6825] [Help]
Hi, thanks, tailwind rocks. I am struggling to setup tailwind as described here: https://tailwindcss.com/docs/guides/create-react-app

I'm on Linux with Node 17.3. I'm trying to get Tailwind, CRA 5, Typescript to work. I've tried many things but Tailwind styles don't get applied.

Here's one thing I've tried:

1. `npx create-react-app@5.0.0 --template typescript web`
2. Follow the setup instructions on the above TailwindCSS website link

I don't know how to troubleshoot this further. Does Typescript have anything to do with it? I also tried cloning [tailwind-cra-next](https://github.com/adamwathan/tailwind-cra-next). It worked after re-installing node modules, but after I added Typescript, it broke again. Specifically I followed the instructions [here](https://create-react-app.dev/docs/adding-typescript/) for adding TS to existing project, and I copied over the default tsconfig.json and src/reactsomething.d.ts from a different CRA 5 project. After those steps, the Tailwind styles stopped appearing.

I also tried using CRA 4 and the old setup method (which has worked perfectly for me many times :man_dancing:), but I get an error message saying to use CRA 5.

(I even tried using an older known good setup, but I think Node 17.3 upgrade or something else broke it.)

## UPDATE 1

I fixed an NPX cache issue and tried the CRA 5 with typescript strategy again. Still not seeing Tailwind styles. I also tried removing Typescript (delete node_modules, remove the ts dependencies and tsconfig.json, change extensions to .js, restart dev server), but I was still getting the same result.

The npx cache issue was like this. `npx create-react-app *` would say I need to remove create-react-app 4.0.3. However when I run the npm -g uninstall command, its not installed. When I list global npm packages, it also says create-react-app isn't installed. I ran this command `rm -rf ~/.npm/_npx` and tried again. I confirmed it made a difference because CRA now generates App.ts as a hook, whereas before it generates a class based component.

## UPDATE 2

After fixing the CRA/npx cache issue, I re-confirmed that **adding Typescript to [tailwind-cra-next](https://github.com/adamwathan/tailwind-cra-next) doesn't work**. I followed the same steps as above.

## UPDATE 3

I was able to get some working Tailwind setup using an old CRA 4 repo. I fixed Node 17 problems by re-installing and upgrading latest deps in my CRA 4 project (see [here](https://stackoverflow.com/questions/69693907/error-err-package-path-not-exported-package-subpath-lib-tokenize-is-not-d)). This will get me going, but I still don't have any way to upgrade to CRA 3.

**Comment by adamwathan** [comment]
Can you share an example project that isn't working? It's by far the most helpful thing you can do to help us troubleshoot 👍

**Comment by adamwathan** [comment]
I followed all of the steps in our documentation but used `--template typescript` like you described and it works as expected:

https://github.com/adamwathan/tw3-cra-typescript-example

Check that repo out and see if there's any difference between what's there and what you've done.

**Comment by srcreigh** [comment]
Thanks Adam, this was my problem - 2 content keys

```js
module.exports = {
  content: [
    "./src/**/*.{js,jsx,ts,tsx}",
  ],
  content: [],
  theme: {
    extend: {},
  },
  plugins: [],
}
```

**Comment by srcreigh** [comment]
When I tried to convert your tailwind-cra-next repo to typescript, I didn't add ts,tsx there. The tailwind-cra-next tailwind config just has js,jsx.

Anyway, it's all working for me now, thanks for your efforts :grin:
